File: drivers/Connector-J/src/main/java/com/actiontech/cleanUp.java
/*
 * Copyright (C) 2016-2020 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
package com.actiontech;

import java.io.*;
import java.sql.Connection;
import java.sql.Statement;

public class cleanUp {

    public static void rmfile(String filename){
        String curpath = System.getProperty("user.dir");
        String path = curpath +  File.separator + filename;
        File loadfile = new File(path);
        if(loadfile.exists()){
            loadfile.delete();
        }
    }

    public static void closeFileWriter(FileWriter fw){
        if(fw!=null){
            try {
                fw.close();

            } catch (Exception fe) {
                System.out.println("close：" + fw + "failed！");
                fe.printStackTrace();
            }
        }
        fw = null;
    }

    public static void closeBufferedWriter(BufferedWriter bw){
        if(bw!=null){
            try {
                bw.close();

            } catch (Exception fe) {
                System.out.println("close：" + bw + "failed！");
                fe.printStackTrace();
            }
        }
        bw = null;
    }

    public static void closeFileReader(FileReader fr){
        if(fr!=null){
            try {
                fr.close();

            } catch (Exception fe) {
                System.out.println("close：" + fr + "failed！");
                fe.printStackTrace();
            }
        }
        fr = null;
    }

    public static void closeBufferedReader(BufferedReader br){
        if(br!=null){
            try {
                br.close();

            } catch (Exception fe) {
                System.out.println("close：" + br + "failed！");
                fe.printStackTrace();
            }
        }
        br = null;
    }

    public static void closeConn(Connection conn){
        if(conn!=null){
            try {
                conn.close();

            } catch (Exception fe) {
                System.out.println("close：" + conn + "failed！");
                fe.printStackTrace();
            }
        }
        conn = null;
    }

    public static void closeStmt(Statement stmt){
        if(stmt!=null){
            try {
                stmt.close();

            } catch (Exception fe) {
                System.out.println("close：" + stmt + "failed！");
                fe.printStackTrace();
            }
        }
        stmt = null;
    }

}

File: drivers/Connector-.Net/netdriver/CleanUp.cs
/*
 * Copyright (C) 2016-2020 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using MySql.Data.MySqlClient;

namespace netdriver
{
    class CleanUp
    {

        public static void Rmfile(String filename)
        {
            String curpath = Environment.CurrentDirectory;
            //String path = curpath + "\\" + filename;
            String path = Path.Combine(curpath, filename);
            if (File.Exists(path))
            {

                try
                {
                    File.Delete(path);
                    Console.WriteLine("Remove file:" + path + " success");
                }
                catch(IOException ioe)
                {
                    Console.WriteLine(ioe.Message);
                    Environment.Exit(-1);
                }  
            }
        }

        public static void CloseStreamReader(StreamReader sr)
        {
            if (sr != null)
            {
                try
                {
                    sr.Close();
                }
                catch (IOException e) {
                    Console.WriteLine(e.ToString());
                    Console.WriteLine("Close " + sr + " failed！");
                    Environment.Exit(-1);
                }
            }
        }
        public static void CloseStreamWriter(StreamWriter sw)
        {
            if (sw != null)
            {
                try
                {
                    sw.Close();
                }
                catch (IOException e)
                {
                    Console.WriteLine(e.ToString());
                    Console.WriteLine("Close " + sw + " failed！");
                    Environment.Exit(-1);
                }
            }
        }

        public static void CloseConnection(MySqlConnection conn)
        {
            if (conn != null)
            {
                try
                {
                    conn.Close();
                }
                catch (IOException e)
                {
                    Console.WriteLine(e.ToString());
                    Console.WriteLine("Close " + conn + " failed！");
                    Environment.Exit(-1);
                }
            }
        }

    }
}
